### PR TITLE
Fixes for agency names 

### PR DESCRIPTION
**Pull request by hlmyhlm**
- Agency abbreviation in header
- Agency full name in description
- Agency full name in dropdown list
- Restored agency's URL mapping
- Refined agency's translation BM/Eng
- Dropdown list width to fit the agency name's length
- Display agency full name (data sources)

### CODE REVIEW

File: packages/datagovmy-ui/src/lib/constants.ts [hunk 1]
```diff
@@ -241,12 +252,106 @@ export const AgencyLink: Record<Agency, string> = {
...
+  "span": "https://www.span.gov.my/",
+  "st": "https://www.st.gov.my/",
+  "swk-energy": "https://www.sarawakenergy.com/",
   "spr": "https://www.spr.gov.my/",
   "ssm": "https://www.ssm.com.my/Pages/Home.aspx",
+  "tnb": "https://www.tnb.com.my/",
   "unhcr": "https://www.unhcr.org/my/",
 } as const;
+
+export const SOURCE_TRANSLATIONS: Record<string, { "ms-MY": string; "en-GB": string }> = {
```
Comment (amalasyrafs) [inline packages/datagovmy-ui/src/lib/constants.ts:276]: since we have agency name translation in `meta` repo, why is this needed here ?

File: packages/datagovmy-ui/src/data-catalogue/Main/index.tsx [hunk 2]
```diff
@@ -43,18 +43,27 @@ interface CatalogueIndexProps {
...
-  const { t } = useTranslation(["catalogue", "common"]);
+  const { t, i18n } = useTranslation(["catalogue", "common"]);
   const scrollRef = useRef<Record<string, HTMLElement | null>>({});
   const filterRef = useRef<CatalogueFilterRef>(null);
   const { size } = useContext(WindowContext);
+  const language = i18n.language; // Get the current language
+
   const sourceOptions = sources.map(source => ({
-    label: source,
+    label: getSourceTranslation(source, language as "ms-MY" | "en-GB"),
```
Comment (amalasyrafs) [inline packages/datagovmy-ui/src/data-catalogue/Main/index.tsx:66]: can probably do this instead, `label: t(`agencies:${source.toLowerCase()}.full`),`

File: packages/datagovmy-ui/src/data-catalogue/Main/index.tsx [hunk 3]
```diff
@@ -72,31 +81,36 @@ const CatalogueIndex: FunctionComponent<CatalogueIndexProps> = ({
   const getHeaderText = (
     site: SiteName
   ): { category: string; description: string; agency: Agency } => {
+    const agencyKey = query.source ? query.source.toLowerCase() : "govt";
+    const agencyTranslation = query.source
+      ? getSourceTranslation(query.source, language as "ms-MY" | "en-GB")
```
Comment (amalasyrafs) [inline packages/datagovmy-ui/src/data-catalogue/Main/index.tsx:86]: same here

File: packages/datagovmy-ui/src/data-catalogue/Main/index.tsx [hunk 4]
```diff
@@ -121,17 +135,24 @@ const CatalogueIndex: FunctionComponent<CatalogueIndexProps> = ({
...
-              width="w-64"
+              width="w-fit"
               placeholder={t("placeholder.source")}
               anchor="left"
               options={sourceOptions}
-              selected={query.source ? { label: query.source, value: query.source } : undefined}
+              selected={
+                query.source
+                  ? {
+                      label: getSourceTranslation(query.source, language as "ms-MY" | "en-GB"),
```
Comment (amalasyrafs) [inline packages/datagovmy-ui/src/data-catalogue/Main/index.tsx:151]: here

File: packages/datagovmy-ui/src/data-catalogue/Show/index.tsx [hunk 5]
```diff
@@ -265,7 +266,9 @@ const CatalogueShow: FunctionComponent<CatalogueShowProps> = ({
                 fields: data.fields,
                 last_updated: data.last_updated,
                 next_update: data.next_update,
-                data_source: data.data_source,
+                data_source: data.data_source.map(
+                  abbr => SOURCE_TRANSLATIONS[abbr]?.[i18n.language as "ms-MY" | "en-GB"] || abbr
```
Comment (amalasyrafs) [inline packages/datagovmy-ui/src/data-catalogue/Show/index.tsx:270]: here

File: packages/datagovmy-ui/src/data-catalogue/Main/index.tsx [hunk 6]
```diff
@@ -72,31 +81,36 @@ const CatalogueIndex: FunctionComponent<CatalogueIndexProps> = ({
...
+            agency: agencyTranslation,
             context: query.source ? "agency" : "",
           }),
-          agency: "govt",
+          agency: agencyKey as Agency,
         };
       case "opendosm":
         return {
           category: t("header_category_agency", {
-            agency: t("agencies:dosm.abbr"),
```
Comment (amalasyrafs) [inline packages/datagovmy-ui/src/data-catalogue/Main/index.tsx:88]: why did u change this ?

File: packages/datagovmy-ui/src/lib/constants.ts (hunk 1 above)
Comment (hlmyhlm) [reply packages/datagovmy-ui/src/lib/constants.ts:276]: Mapped the key to full translation. Earlier was told not to touch the BE part.

File: packages/datagovmy-ui/src/data-catalogue/Main/index.tsx (hunk 2 above)
Comment (hlmyhlm) [reply packages/datagovmy-ui/src/data-catalogue/Main/index.tsx:66]: Ok

File: packages/datagovmy-ui/src/data-catalogue/Main/index.tsx (hunk 3 above)
Comment (hlmyhlm) [reply packages/datagovmy-ui/src/data-catalogue/Main/index.tsx:86]: will check

File: packages/datagovmy-ui/src/data-catalogue/Main/index.tsx (hunk 4 above)
Comment (hlmyhlm) [reply packages/datagovmy-ui/src/data-catalogue/Main/index.tsx:151]: will check

File: packages/datagovmy-ui/src/data-catalogue/Show/index.tsx (hunk 5 above)
Comment (hlmyhlm) [reply packages/datagovmy-ui/src/data-catalogue/Show/index.tsx:270]: will relook

File: packages/datagovmy-ui/src/data-catalogue/Main/index.tsx (hunk 6 above)
Comment (hlmyhlm) [reply packages/datagovmy-ui/src/data-catalogue/Main/index.tsx:88]: Will relook

File: packages/datagovmy-ui/src/lib/constants.ts (hunk 1 above)
Comment (amalasyrafs) [reply packages/datagovmy-ui/src/lib/constants.ts:276]: its not the BE part. its the `meta` part.

**Comment by amalasyrafs** [comment]
tldr: to not use `SOURCE_TRANSLATIONS` (remove it) and use translation from `meta`. its better to keep single source of truth for the translations.

File: packages/datagovmy-ui/src/lib/constants.ts (hunk 1 above)
Comment (hlmyhlm) [reply packages/datagovmy-ui/src/lib/constants.ts:276]: ok

**Comment by hlmyhlm** [comment]
> tldr: to not use `SOURCE_TRANSLATIONS` (remove it) and use translation from `meta`. its better to keep single source of truth for the translations.

What about the URL (portal link for agencies)? Should I add into meta?

**Comment by amalasyrafs** [comment]
> > tldr: to not use `SOURCE_TRANSLATIONS` (remove it) and use translation from `meta`. its better to keep single source of truth for the translations.
> 
> Alright. What about the URL (portal link for agencies)? Should I add into meta?

no that one, tak perlu. bcs all the links mmg in the `front` repo, whereas the `agencies` name are inside the `meta` repo.
